openedx/adg/lms/applications/api_views.py

```diff
@@ -0,0 +1,221 @@
+"""
+API views for applications app
+"""
+
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from django.utils.translation import ugettext_lazy as _
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from rest_framework import status
+from rest_framework.authentication import BasicAuthentication
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from .models import Education, WorkExperience
+from .serializers import EducationSerializer, WorkExperienceSerializer
+
+
+class EducationAPIView(APIView):
+    """
+    CRUD implementation of education data
+    """
+    # TODO: BasicAuthentication is added here for testing purposes, pleases remove this when these APIs are integrated.
+    authentication_classes = (BasicAuthentication, SessionAuthenticationAllowInactiveUser,)
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request):
+        """
+        Retrieve complete education record for requested user
+
+        Arguments:
+            request (HttpRequest): Django request object
+
+        Returns:
+            HttpResponse: Education records in JSON format
+
+        """
+        education_records = Education.objects.filter(user_application__user=request.user)
+        serializer = EducationSerializer(education_records, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+        """
+        Create education record for requested user
+
+        Arguments:
+            request (HttpRequest): Django request object
+
+        Returns:
+            HttpResponse: Newly created education records in JSON format
+
+        """
+        serializer = EducationSerializer(data=request.data, many=True)
+
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def patch(self, request, pk):
+        """
+        Update education record for requested user
+
+        Arguments:
+            request (HttpRequest): Django request object
+            pk (Int): Id of education record that needs to be updated
+
+        Returns:
+            HttpResponse: Updated education record in JSON format
+
+        """
+        try:
+            education = get_object_or_404(Education.objects.all(), pk=pk)
+        except Http404:
+            return get_response(
+                _('Education record with id `{pk}` doesn\'t exists.').format(pk=pk),
+                status.HTTP_404_NOT_FOUND
+            )
```

Comment:
if we have to use try catch please directly use `get` method and if it raise DoesNotExist exception you can return your response. 

`get_object_or_404` is used when we need to avoid try catch with get and want to directly return 404